apps/publish/content/common.py

```diff
@@ -861,6 +860,7 @@ def _publish_associated_items(self, original, updates=None):
                         remove_unwanted(association_updates)
                         publish_service.patch(id=association_updates.pop(config.ID_FIELD), updates=association_updates)
 
+            insert_into_versions(doc=associated_item)
```

Comment:
can you add  a comment when this should be actually called? there are too many ifs..

Reply:
sure.

Reply:
actually not sure why it's needed, if the item is being published now it would get the version saved automatically, wouldn't it? and if it was published already it's not modified so there should be no need to put it there

Reply:
if the item is being published it would get the version saved automatically, but it will not save/update the version of the associated item in the archive_version, that's why we need to add it here.